components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadsFeature.kt

```diff
@@ -191,12 +192,15 @@ class DownloadsFeature(
             useCases.consumeDownload.invoke(tab.id, download.id)
         }
 
-        if (!isAlreadyADialogCreated() && fragmentManager != null) {
-            dialog.show(fragmentManager, FRAGMENT_TAG)
+        if (!dialog.isAdded && !isAlreadyADialogCreated() && fragmentManager != null) {
```

Comment:
Fix looks OK. Were you able to reproduce the original crash at all? Do you have STRs?

Reply:
(Haven't been able to repro the crash yet and would like to verify manually as well).

Reply:
We don't have the STR, but I think the issue was caused because we were calling [dialog.show](https://developer.android.com/reference/androidx/fragment/app/DialogFragment#show(androidx.fragment.app.FragmentManager,%20java.lang.String)) that doesn't warranty the  fragment transaction will be committed right away , allowing to dialog to re-add, for this reason we are using [dialog.showNow](https://developer.android.com/reference/androidx/fragment/app/DialogFragment#showNow(androidx.fragment.app.FragmentManager,%20java.lang.String)) that adds the fragment right away and guarding the call using `dialog.isAdded`.

Reply:
Not sure how to replicate, but it looks like a race condition, where we added a previous dialog, but it wasn't commit when line 195 was called

Reply:
Yes, understood the patch, but was hoping we can reproduce to make sure.

Reply:
I'll look into the memory leak and see if I can "simulate" this as well somehow.

Reply:
👍

Reply:
One thing that I noticed, was that most of the reported devices are low memory phones, not sure if it's connected

Reply:
OK, we verifed this by overriding show and delaying the `commit`. The solution works but we're gonna fold it into the fix for https://github.com/mozilla-mobile/android-components/issues/7180

Reply:
👍